File: backend/src/schemas/story.py
from datetime import datetime

from pydantic import BaseModel

from src.schemas.article import ArticleRead


class StoryBase(BaseModel):
    headline: str
    summary: str | None = None
    published_at: datetime


class StoryRead(StoryBase):
    id: int
    left_count: int
    center_count: int
    right_count: int
    is_blindspot: bool
    blindspot_perspective: str | None = None

    model_config = {"from_attributes": True}


class StoryWithCoverage(StoryRead):
    total_coverage: int = 0
    image_url: str | None = None


class StoryDetailResponse(StoryRead):
    articles: list[ArticleRead] = []

File: mobile/types/story.ts
export interface Source {
  id: number;
  name: string;
  domain: string;
  bias_rating: string;
  confidence: number;
}

export interface Article {
  id: number;
  story_id: number;
  source_id: number;
  title: string;
  url: string;
  published_at: string | null;
  snippet: string | null;
  image_url: string | null;
  source: Source | null;
}

export interface Story {
  id: number;
  headline: string;
  summary: string | null;
  published_at: string;
  left_count: number;
  center_count: number;
  right_count: number;
  is_blindspot: boolean;
  blindspot_perspective: string | null;
  total_coverage: number;
  image_url: string | null;
}

export interface StoryDetail extends Story {
  articles: Article[];
}
